src/detectors/LOWQ2/LOWQ2.cc

```diff
@@ -30,6 +32,31 @@ extern "C" {
          app
     ));
 
+    // This should really be done before digitization as summing hits in the same cell couldn't evet be mixed between layers. At the moment just prep for clustering.
+    std::vector<int> moduleIDs{1,2};
+    std::vector<int> layerIDs {0,1,2,3};
+    std::vector<std::vector<int>> segmentIDs{};
+    std::vector<std::string> segmentDiv;
+
+    for(int mod_id : moduleIDs){
+      for(int lay_id : layerIDs){
+        segmentIDs.push_back({mod_id,lay_id});
+        segmentDiv.push_back(fmt::format("TaggerTrackerM{}L{}RawHits",mod_id,lay_id));
+      }
+    }
+
+    app->Add(new JOmniFactoryGeneratorT<SplitGeometry_factory<edm4eic::RawTrackerHit>>(
+         "TaggerTrackerSplitHits",
+         {"TaggerTrackerRawHits"},
+         segmentDiv,
+         {
+           .divisions = segmentIDs,
+           .readout   = "TaggerTrackerHits",
+           .division  = {"module","layer"},
+         },
+         app
+      )
+    );
```

Comment:
I would happily make it more generic but can't currently picture how you're suggesting to go about this.

Reply:
This will need to be left for to a future PR.

Were you envisioning defining the lambda function in (e.g.) LOWQ2.cc and passing that to a factory constructor? This feels like it's moving part of what should be contained inside an algorithm to the wrong place and I haven't worked out a way of passing an appropriate lambda into the factory.

Options this should presumably be able to cope (amongst others) with:
A collection member potentially being a member of multiple subset collections
Dividing the collection up based on ranges rather than discrete values
The geometry splitting application requires passing the detector segmentation decoder

Anyway got myself a bit tangled up thinking about it.

Reply:
It's ugly but appears to work by passing the lambda function through the algorithm config. A couple of examples covering my previous use cases here:
[https://github.com/eic/EICrecon/blob/generalise-splitgeometry-to-divide-any-collection-by-lambda/src/detectors/LOWQ2/LOWQ2.cc](https://github.com/eic/EICrecon/blob/generalise-splitgeometry-to-divide-any-collection-by-lambda/src/detectors/LOWQ2/LOWQ2.cc)

Reply:
Fixed in a way I'm more or less happy with (thanks somewhat to copilot). The SubDivideCollection algorithm now just gets passed a std::function through the config returning indices for which collections an entry should be included in. I've introduced a couple of templated examples as functors into a separate header file.